### PR TITLE
feat(curriculum): add to-do list lab

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+The links on the page should have no underline by default.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.todo-list li a').getPropVal('text-decoration'), 'none' || new __helpers.CSSHelp(document).getStyle('a').getPropVal('text-decoration'), 'none');
+```
+
+The links should change color when hovered over.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.sub-item-link:link')?.color || new __helpers.CSSHelp(document).getStyle('a:link')?.color);
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:127]: Shouldn't it be `a:hover`?

**Review by Dario-DC — COMMENTED** [review]
A couple of tests are not working as expected. I left a few comments.

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+assert(!!anchors.length);
+for (let anchor of anchors) {
+    assert.equal(anchor.target, '_blank');
+}
+```
+
+The links on the page should have no underline by default.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.todo-list li a').getPropVal('text-decoration'), 'none' || new __helpers.CSSHelp(document).getStyle('a').getPropVal('text-decoration'), 'none');
```
[the remark this answers is not in the document]
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md]: @Dario-DC 
This test didn't work.

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+7. When the links are clicked, the color should change to another colour of your choice.
+    
+# --hints--
+
+You should have an unordered list with the class `todo-list`.
+
+```js
+assert.exists(document.querySelector('.todo-list'));
+```
+
```
[the remark this answers is not in the document]
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:39]: done!

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+All of the `input` elements should have an `id`
+
+```js
+const inputs = document.querySelectorAll('input');
+for (let input of inputs) {
+    assert.notEmpty(input?.id);
+}
+```
+
+After the `label` elements, there should be an unordered list with the class `sub-item`.
```
[the remark this answers is not in the document]
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:68]: I have left this hint as it is, but made the test a bit specific:

Now, `label + ul.sub-item` would select all `ul` elements that:
- Directly follow a `label` element in the DOM.
- Have the class `sub-item`.

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md]: I updated the test, but it has room for improvement.

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+challengeType: 14
+dashedName: build-a-stylized-to-do-list
+demoType: onClick
+---
+
+# --description--
+
+In this lab, you will practice the different styles that can be applied to links when they are hovered over, focused, clicked, and visited.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. **Do not copy this demo project**.
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:13]:
```suggestion
**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.
```

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+   - An unordered list with the class `sub-item`.
+   - A list item with an `anchor` element with the class `sub-item-link` and a valid `href` within the `sub-item` unordered list.
+  
+4. There shouldn't be any list styles applied to the unordered lists.
+5. The anchor elements shouldn't have any default text decoration.
+6. You should set the text color of the links to a colour of your choice.
+7. When the links are visited, the color should change to another colour of your choice.
+8. When the links are hovered over, the color should change to another colour of your choice.
+9. When the links are focused, there should be a coloured outline around the link.
+10. When the links are clicked, the color should change to another colour of your choice.
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:32]:
```suggestion
4. You shouldn't apply any list styles to the unordered lists.
5. Your anchor elements shouldn't have any default text decoration.
6. You should set the text color of the links to a color of your choice.
7. When your links are visited, the color should change to another color of your choice.
8. When your links are hovered over, the color should change to another color of your choice.
9. When your links are focused, there should be a colored outline around the link.
10. When your links are clicked, the color should change to another color of your choice.
```

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+**User Stories:**
+
+1. You should have an unordered list with the class `todo-list`.
+2. Inside the unordered list, you should have 4 list items.
+3. Inside each list item, there should be:
+   
+   - An `input` element with the type `checkbox` and `id` set to a unique value.
+   - A `label` element with the `for` attribute set to the corresponding `input` element's `id`.
+   - An unordered list with the class `sub-item`.
+   - A list item with an `anchor` element with the class `sub-item-link` and a valid `href` within the `sub-item` unordered list.
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:24]:
```suggestion
   - A list item with an `anchor` element in it. The `anchor should have the class `sub-item-link`, a valid `href` value, and a `target` value that makes the link open in a new tab.
```

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.sub-item-link:focus')?.outline || new __helpers.CSSHelp(document).getStyle('a:focus')?.outline);
+```
+
+The links should change color once visited.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.sub-item-link:visited')?.color || new __helpers.CSSHelp(document).getStyle('a:visited')?.color);
+```
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:161]: These tests all require using the `.sub-item-link` selector. So if I just use `a` or `a.sub-item-link`, the tests don't pass. We should either change the text to specifically say to use the `.sub-item-link` selector (might want to add it to the user stories if we go that route), or make the tests more flexible.

**Review by moT01 — COMMENTED** [review]
This is looking pretty good @zairahira 👍 

I left a few grammar suggestions on the user stories and had an issue with a number of the tests.

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md (hunk 8 above)
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:161]: I have made the tests flexible by targeting `a.sub-item-link`, `.sub-item-link`, and `a`.

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+# --description--
+
+In this lab, you will practice the different styles that can be applied to links when they are hovered over, focused, clicked, and visited.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories:**
+
+1. You should have an unordered list with the class `todo-list`.
+2. Inside the unordered list, you should have 4 list items.
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:18]:
```suggestion
2. Inside the unordered list, you should have four list items.
```

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+3. Inside each list item, there should be:
+   
+   - An `input` element with the type `checkbox` and `id` set to a unique value.
+   - A `label` element with the `for` attribute set to the corresponding `input` element's `id`.
+   - An unordered list with the class `sub-item`.
+   - A list item with an `anchor` element in it. The `anchor should have the class `sub-item-link`, a valid `href` value, and a `target` value that makes the link open in a new tab.
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:24]:
```suggestion
   - A list item with an anchor element in it. The anchor should have the class `sub-item-link`, a valid `href` value, and a `target` value that makes the link open in a new tab.
```

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md [hunk 11]
```diff
@@ -0,0 +1,320 @@
...
+  
+4. You shouldn't apply any list styles to the unordered lists.
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:26]: I'm not sure what this one is supposed to mean. Does it mean I should use `list-style: none;` or am I not supposed to change the default list-style. If so, I would reword this to something like "You should remove the default list style from your unordered lists."

And we could add a test for it if so. Also, is it supposed to apply to all the UL's? or just the main UL? I would make it so it's just the main one. "You should remove the default list style from your `.todo-list`"

**Review by moT01 — COMMENTED** [review]
It's looking good. Found a few more things...

File: curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md (hunk 11 above)
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md:26]: This instruction is supposed to remove the default style from the main unordered list. The campers can add any style to the other list inside and it is optional.

I have updated the instruction and added a new test.
It works for both:

```css
.todo-list {
    list-style: none;
}
```

```css
.todo-list {
    list-style-type: none;
}
```

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Review by jdwilkin4 — APPROVED** [review]
